### PR TITLE
Added congratulation controller

### PR DESCRIPTION
**Pull request by AlexKievDeveloper**
1. Removed ServiceLocator, DataSourceFactiory, PropertyReader, CardServlet, GetCardsServlet, CongratulationServlet, LoginLogautServlet, RegistrationServlet and tests for them.
2. Added ExceptionHandlingController.
3. Added CongratulationController and tests for it.
4. Changed pom.xml (added dependencies).
5. Changed flywayConfig (added dispatcher-servlet as imported resource).
6. Changed application context.
7. Changed DefaultCongratulationService, JdbcCongratulationDao.

### CODE REVIEW

**Review by Yuliia-Sh — APPROVED** [review]

**Review by RomanKlishch — APPROVED** [review]

File: src/main/java/com/greetingcard/dao/jdbc/JdbcCardDao.java [hunk 1]
```diff
@@ -1,6 +1,6 @@
-package com.greetingcard.dao.jdbc;
 
-import com.greetingcard.ServiceLocator;
+/*package com.greetingcard.dao.jdbc;
+
 import com.greetingcard.dao.CardDao;
 import com.greetingcard.dao.CongratulationDao;
 import com.greetingcard.dao.jdbc.mapper.CardAndCongratulationRowMapper;
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/dao/jdbc/JdbcCardDao.java:6]: Why didn't you delete comment?

File: src/main/java/com/greetingcard/security/DefaultSecurityService.java [hunk 2]
```diff
@@ -83,3 +86,5 @@ String getHashPassword(String saltAndPassword) {
     }
 
 }
+
+*/
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/security/DefaultSecurityService.java:90]: Why didn't you delete comment?

File: src/main/java/com/greetingcard/web/controller/ExceptionHandlingController.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import java.util.Map;
+
+@RestControllerAdvice
+public class ExceptionHandlingController {
+
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/web/controller/ExceptionHandlingController.java:14]: Why did you add this class?

File: src/main/java/com/greetingcard/web/controller/UserController.java [hunk 4]
```diff
@@ -1,47 +1,52 @@
...
+
+                response.setStatus(HttpServletResponse.SC_CREATED);
+
+            } catch (RuntimeException e) {
+                response.getWriter().print(e.getMessage());
+                response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+                log.error("Exception while registration user login: {}", user.getLogin());
+            }*//*
+
+
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/web/controller/UserController.java:50]: )

File: src/main/java/com/greetingcard/web/controller/ExceptionHandlingController.java (hunk 3 above)
Comment (AlexKievDeveloper) [reply src/main/java/com/greetingcard/web/controller/ExceptionHandlingController.java:14]: > Why did you add this class?

I added this class because it is part of the error handling system we discussed this morning.

**Review by RomanKlishch — APPROVED** [review]

**Review by OleksandrGubariev — APPROVED** [review]
